File: javascript/etl/etl.js
var ETL = function() {};

ETL.prototype.transform = function(oldScore) {
  var newScore = {};
  for (var pointValue in oldScore) {
    for (var i = 0; i < oldScore[pointValue].length; i++) {
      var letter = oldScore[pointValue][i].toLowerCase();
      newScore[letter] = parseInt(pointValue);
    };
  };
  return newScore;
};

module.exports = ETL;

File: java/etl/src/main/java/Etl.java
import java.util.List;
import java.util.Map;
import java.util.HashMap;

public class Etl {
  public Map<String, Integer> transform(Map<Integer, List<String>> old) {
    Map<String, Integer> newScores = new HashMap<String, Integer>();
    for (Map.Entry<Integer, List<String>> entry : old.entrySet()) {
      for (String oldLetter : entry.getValue()) {
        String newLetter = oldLetter.toLowerCase();
        newScores.put(newLetter, entry.getKey());
      }
    }
    return newScores;
  }
}
